lib/mongoid/criteria.rb

```diff
@@ -555,8 +555,7 @@ def merge_type_selection
     # @since 3.0.3
     def type_selectable?
       klass.hereditary? &&
-        !selector.keys.include?("_type") &&
-        !selector.keys.include?(:_type)
+        !selector.keys.include?(self.discriminator_key)
```

Comment:
This still needs to account for string and symbol keys.

Reply:
I fixed this, but I don't understand why it was needed. `self.discriminator_key` is going to equal whatever the user set the discriminator key, whether it's a symbol or a string, no?